File: entity_framework/lab_1/Supplier.cs
// public class Supplier {
//     public int SupplierId { get; set; }
//     public required String CompanyName { get; set; }
//     public String? Street { get; set; }
//     public String? City { get; set; }
//     // public List<Product>? Products { get; set; }
//     public ICollection<Product>? Products { get; set; }
    
// }

public class Supplier : Company
{
    public string? BankAccountNumber { get; set; }
    public ICollection<Product>? Products { get; set; }
}

File: hibernate/src/main/java/org/example/Supplier.java
package org.example;

import jakarta.persistence.*;

import java.util.List;

@Entity
public class Supplier {
    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    private int supplierID;
    private String companyName;
    private String street;
    private String city;
    @OneToMany(mappedBy = "supplier")
    private List<Product> products;
    public Supplier() {
    }
    public Supplier(String companyName, String street, String city) {
        this.companyName = companyName;
        this.street = street;
        this.city = city;
    }
    public Supplier(String companyName, String street, String city, List<Product> products) {
        this.companyName = companyName;
        this.street = street;
        this.city = city;
        this.products = products;
    }
    public void setProducts(List<Product> products) {
        this.products = products;
    }
}
